addon/components/pix-select.js

```diff
@@ -155,11 +155,11 @@ export default class PixSelect extends Component {
   focus() {
     if (this.isExpanded) {
       if (this.args.value) {
-        this.rootElement.querySelector("[aria-selected='true']").focus();
+        this.rootElement.querySelector("[aria-selected='true']")?.focus();
```

Comment:
mais j'ai l'impression que les tests vont rester flaky vu qu'on va pas trouver l'élément, donc on va rien focus ?

Reply:
Oui il faudrait pouvoir lancer les tests concernés avec cette branch de pix-ui pour être sûr 😕

Reply:
Question, est-ce util d'avoir une promise pour chercher le rootElement dans le constructor. sachant que lorsque l'on fait l'action focus . ( il existe forcément dans le DOM ce selecteur non ? )

Est ce qu'un simple

`document.getElementById('mon-selector).querySelect(...).focus()` suffirait ?

Reply:
> Question, est-ce util d'avoir une promise pour chercher le rootElement dans le constructor. sachant que lorsque l'on fait l'action focus . ( il existe forcément dans le DOM ce selecteur non ? )

Le message d'erreur `Cannot read properties of null (reading 'focus')` indique que le rootElement existe bien à ce moment-là, sinon il nous aurait dit `Cannot read properties of null (reading 'querySelector')`. L'erreur serait donc liée au `[aria-selected='true']` qui n'est pas toujours actualisé à temps dans le DOM (comme l'avait dit @bpetetot sur Slack).

Je soupçonne que les tests qui sont flaky aujourd'hui passaient seulement parce qu'il y avait d'autres éléments `[aria-selected='true']` dans la page testée, car le sélecteur n'était pas scopé à l'instance du composant. Dans ce cas, le focus n'était pas important au déroulement de ces tests, vu qu'on focusait potentiellement le mauvais élément 🫣.

Néanmoins, afin d'avoir un comportement correct, on pourrait donner à chaque option du select un id basé sur sa valeur. 
On récupérerait alors l'option à partir de cet id au lieu de son état `aria-selected` :
```
this.rootElement.getElementById(${this.listId}-${this.args.value}`).focus()
```
L'élément ne devrait jamais être null dans ce cas, sauf si la valeur passée en props est erronée.

Qu'en pensez-vous?

Reply:
@fael-b @xav-car @laura-bergoens 
Je pense qu'il ne faut pas se baser sur `this.rootElement`, on a aucune garantie qu'il soit correctement rendu au moment où on l'utilise (surtout dans les tests qui s'exécutent très rapidement.

Dans l'idéal, il faudrait enlever un maximum l'utilisation de `rootElement`. Par exemple, dans ce cas on peut se baser sur l'évènement de `focus` et l'écrire de :

```js
@action
focus(event) {
  if (!event.target) return;
  if (!this.isExpanded) return;

  if (this.args.value) {
    event.target.querySelector("[aria-selected='true']")?.focus();
  } else if (this.args.isSearchable) {
    event.target.querySelector(`#${this.searchId}`).focus();
  } else if (this.displayDefaultOption) {
    event.target.querySelector("[aria-selected='true']")?.focus();
  }
}
```

Et je pense qu'on devrait éviter l'utilisation de ce `rootElement` quand on peut et plutôt utiliser les `events` des actions.

Reply:
Effectivement, le rootElement n'est peut-être pas ce qui cause les flaky aujourd'hui, mais rien ne nous garantie qu'il n'en causera pas dans le futur.
On peut partir sur ta solution et carrément retirer le rootElement (il n'est utilisé que pour l'action focus) si ça convient à tout le monde 👍

Reply:
À mon avis, la façon dont est récupéré le `rootElement` est la cause des flakies.